catalog/dags/data_refresh/remap_table_constraints.py

```diff
@@ -0,0 +1,233 @@
+"""
+# Remap Table Constraints
+
+This file contains TaskGroups related to remapping the table constraints from the live media table onto the temp table to prepare it for promotion.
+
+"""
+
+import logging
+from typing import NamedTuple
+
+from airflow.decorators import task, task_group
+
+from common.sql import run_sql
+from data_refresh import queries
+
+
+logger = logging.getLogger(__name__)
+
+
+class ConstraintInfo(NamedTuple):
+    constraint_table: str
+    constraint_name: str
+    constraint_statement: str
+
+
+def fetch_all_tuples(cursor):
+    try:
+        rows = cursor.fetchall()
+        return [ConstraintInfo(row[0], row[1], row[2]) for row in rows]
+    except Exception as e:
+        raise ValueError("Unable to extract expected row data from cursor") from e
+
+
+def _is_foreign_key(_statement, table):
```

Comment:
Is the proceeding underscore necessary here? Usually that's used to denote private variables

Reply:
Ah, I was confused because I thought you were questioning the preceding underscore in the function name 😅 This was copied directly from the ingestion server and I didn't even notice `_statement`, not sure why it's there! Can definitely be removed.